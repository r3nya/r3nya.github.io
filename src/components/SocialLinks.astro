---
import Icon from './Icon.astro';
import type { Messages } from '../i18n/messages';

interface Props {
  links: Messages['links'];
}

const { links }: Props = Astro.props;
---

<section
  class="col-span-1 grid gap-2"
  role="navigation"
  aria-label="Social links"
  data-section="links"
>
  {
    links.map((link) => (
      <a
        href={link.url}
        class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark"
        aria-label={link.title}
        data-link={link.icon}
        rel="me noopener noreferrer"
        target="_blank"
      >
        <Icon name={link.icon} />
        <span class="group-hover:underline">{link.title}</span>
      </a>
    ))
  }
</section>
