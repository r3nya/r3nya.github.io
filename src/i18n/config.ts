export const locales = ['en', 'ru', 'es'] as const;
export type Locale = (typeof locales)[number];
export const defaultLocale: Locale = 'en';
