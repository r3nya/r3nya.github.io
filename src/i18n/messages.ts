import type { Locale } from './config';

export type Messages = {
  title: string;
  name: string;
  headline: string;
  links: { title: string; icon: 'github' | 'linkedin' | 'paper-plane' | 'twitter'; url: string }[];
};

const commonLinks = {
  github: { title: 'Github', icon: 'github', url: 'https://github.com/r3nya' },
  linkedin: { title: 'LinkedIn', icon: 'linkedin', url: 'https://www.linkedin.com/in/r3nya' },
  telegram: { title: 'Telegram', icon: 'paper-plane', url: 'https://t.me/r3nya' },
  twitter: { title: 'X/Twitter', icon: 'twitter', url: 'https://x.com/r3nya' },
} as const;

export const messages: Record<Locale, Messages> = {
  en: {
    title: 'Andrey Makarov aka r3nya',
    name: 'Andrey Makarov',
    headline: 'Software Developer 👨‍💻',
    links: [commonLinks.github, commonLinks.linkedin, commonLinks.telegram, commonLinks.twitter],
  },
  ru: {
    title: 'Андрей Макаров aka r3nya',
    name: 'Андрей Макаров',
    headline: 'Разработчик ПО 👨‍💻',
    links: [commonLinks.github, commonLinks.linkedin, commonLinks.telegram, commonLinks.twitter],
  },
  es: {
    title: 'Andrey Makarov aka r3nya',
    name: 'Andrey Makarov',
    headline: 'Desarrollador 👨‍💻',
    links: [commonLinks.github, commonLinks.linkedin, commonLinks.telegram, commonLinks.twitter],
  },
};
