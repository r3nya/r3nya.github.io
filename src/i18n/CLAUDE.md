# Internationalization - CLAUDE.md

Internationalization (i18n) system guidelines for Claude Code.

## Overview

This directory contains the complete internationalization system supporting English, Russian, and Spanish locales.

## File Structure

- **`config.ts`** - Locale configuration and types
- **`messages.ts`** - All localized content with TypeScript types

## Supported Locales

- **English** (`en`) - Default locale
- **Russian** (`ru`) - Full translation
- **Spanish** (`es`) - Full translation

## Adding New Content

### 1. Update Types

First, add new keys to the `Messages` type in `messages.ts`:

```typescript
export type Messages = {
  title: string;
  name: string;
  headline: string;
  // Add your new key here
  newContent: string;
  links: {
    title: string;
    icon: 'github' | 'linkedin' | 'paper-plane' | 'twitter';
    url: string;
  }[];
};
```

### 2. Add Translations

Add translations for all supported locales:

```typescript
export const messages: Record<Locale, Messages> = {
  en: {
    title: 'Andrey Makarov aka r3nya',
    name: 'Andrey Makarov',
    headline: 'Software Developer üë®‚Äçüíª',
    newContent: 'Your English content here',
    // ... other content
  },
  ru: {
    title: '–ê–Ω–¥—Ä–µ–π –ú–∞–∫–∞—Ä–æ–≤ aka r3nya',
    name: '–ê–Ω–¥—Ä–µ–π –ú–∞–∫–∞—Ä–æ–≤',
    headline: '–†–∞–∑—Ä–∞–±–æ—Ç—á–∏–∫ –ü–û üë®‚Äçüíª',
    newContent: '–í–∞—à —Ä—É—Å—Å–∫–∏–π –∫–æ–Ω—Ç–µ–Ω—Ç –∑–¥–µ—Å—å',
    // ... other content
  },
  es: {
    title: 'Andrey Mak√°rov aka r3nya',
    name: 'Andrey Mak√°rov',
    headline: 'Desarrollador üë®‚Äçüíª',
    newContent: 'Su contenido en espa√±ol aqu√≠',
    // ... other content
  },
};
```

## Usage in Components

```astro
---
import { messages } from '../i18n/messages';
import type { Locale } from '../i18n/config';

interface Props {
  locale: Locale;
}

const { locale } = Astro.props;
const t = messages[locale];
---

<h1>{t.title}</h1>
<p>{t.newContent}</p>
```

## Routing System

The i18n system works with Astro's dynamic routing:

- Default route: `/` (English)
- Localized routes: `/ru/`, `/es/`
- Invalid locale codes fallback to English

## Best Practices

- **Always provide translations** for all supported locales
- **Use descriptive keys** that indicate content purpose
- **Keep translations consistent** across locales
- **Test with all locales** to ensure proper rendering
- **Maintain TypeScript types** for compile-time validation

## Common Links

Social media links are shared across all locales using the `commonLinks` object to maintain consistency while allowing locale-specific titles.
