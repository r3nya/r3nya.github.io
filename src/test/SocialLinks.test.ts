// @vitest-environment happy-dom
import { describe, it, expect } from 'vitest';
import SocialLinks from '../components/SocialLinks.astro';
import { renderAstroComponent } from './helpers';
import { messages } from '../i18n/messages';

describe('SocialLinks', () => {
  it('renders with English links', async () => {
    const result = await renderAstroComponent(SocialLinks, {
      props: {
        links: messages.en.links,
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });

  it('renders with Russian links', async () => {
    const result = await renderAstroComponent(SocialLinks, {
      props: {
        links: messages.ru.links,
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });

  it('renders with Spanish links', async () => {
    const result = await renderAstroComponent(SocialLinks, {
      props: {
        links: messages.es.links,
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });
});