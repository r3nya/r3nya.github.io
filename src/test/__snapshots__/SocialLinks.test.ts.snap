// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialLinks > renders with English links 1`] = `"<section class="col-span-1 grid gap-2" role="navigation" aria-label="Social links" data-section="links"> <a href="https://github.com/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="Github" data-link="github" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"></path></svg> <span class="group-hover:underline">Github</span> </a><a href="https://www.linkedin.com/in/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="LinkedIn" data-link="linkedin" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z M2 9h4v12H2z M4 6a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"></path></svg> <span class="group-hover:underline">LinkedIn</span> </a><a href="https://t.me/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="Telegram" data-link="paper-plane" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z m7.318-19.539-10.94 10.939"></path></svg> <span class="group-hover:underline">Telegram</span> </a><a href="https://x.com/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="X/Twitter" data-link="twitter" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M22 4s-.7 2.1-2 3.4c1.6 10-9.4 17.3-18 11.6 2.2.1 4.4-.6 6-2C3 15.5.5 9.6 3 5c2.2 2.6 5.6 4.1 9 4-.9-4.2 4-6.6 7-3.8 1.1 0 3-1.2 3-1.2z"></path></svg> <span class="group-hover:underline">X/Twitter</span> </a> </section>"`;

exports[`SocialLinks > renders with Russian links 1`] = `"<section class="col-span-1 grid gap-2" role="navigation" aria-label="Social links" data-section="links"> <a href="https://github.com/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="Github" data-link="github" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"></path></svg> <span class="group-hover:underline">Github</span> </a><a href="https://www.linkedin.com/in/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="LinkedIn" data-link="linkedin" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z M2 9h4v12H2z M4 6a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"></path></svg> <span class="group-hover:underline">LinkedIn</span> </a><a href="https://t.me/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="Telegram" data-link="paper-plane" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z m7.318-19.539-10.94 10.939"></path></svg> <span class="group-hover:underline">Telegram</span> </a><a href="https://x.com/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="X/Twitter" data-link="twitter" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M22 4s-.7 2.1-2 3.4c1.6 10-9.4 17.3-18 11.6 2.2.1 4.4-.6 6-2C3 15.5.5 9.6 3 5c2.2 2.6 5.6 4.1 9 4-.9-4.2 4-6.6 7-3.8 1.1 0 3-1.2 3-1.2z"></path></svg> <span class="group-hover:underline">X/Twitter</span> </a> </section>"`;

exports[`SocialLinks > renders with Spanish links 1`] = `"<section class="col-span-1 grid gap-2" role="navigation" aria-label="Social links" data-section="links"> <a href="https://github.com/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="Github" data-link="github" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"></path></svg> <span class="group-hover:underline">Github</span> </a><a href="https://www.linkedin.com/in/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="LinkedIn" data-link="linkedin" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z M2 9h4v12H2z M4 6a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"></path></svg> <span class="group-hover:underline">LinkedIn</span> </a><a href="https://t.me/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="Telegram" data-link="paper-plane" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z m7.318-19.539-10.94 10.939"></path></svg> <span class="group-hover:underline">Telegram</span> </a><a href="https://x.com/r3nya" class="flex items-center text-2xl font-medium text-palette-5-light group md:text-3xl dark:text-palette-5-dark hover:text-palette-6-light dark:hover:text-palette-6-dark" aria-label="X/Twitter" data-link="twitter" rel="me noopener noreferrer" target="_blank"> <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="w-7 h-7 mr-3 flex-shrink-0"><path d="M22 4s-.7 2.1-2 3.4c1.6 10-9.4 17.3-18 11.6 2.2.1 4.4-.6 6-2C3 15.5.5 9.6 3 5c2.2 2.6 5.6 4.1 9 4-.9-4.2 4-6.6 7-3.8 1.1 0 3-1.2 3-1.2z"></path></svg> <span class="group-hover:underline">X/Twitter</span> </a> </section>"`;
